src/apel_plugin.py

```diff
@@ -145,22 +135,57 @@ def get_submit_host(record, config):
             "%2F", "/"
         )
     except TypeError:
-        logging.warning(
-            f"No submithost found in record {record.record_id}, using default"
-        )
+        logging.warning(f"No submithost found in record {record.record_id}")
+        logging.warning(f"Send default {default_submit_host}")
         submit_host = default_submit_host
 
     return submit_host
 
 
+def get_voms_info(record, config):
+    meta_key_voms = config["auditor"].get("meta_key_voms")
+    voms_dict = {}
+
+    try:
+        voms_string = record.meta.get(meta_key_voms)[0].replace("%2F", "/")
+        voms_list = voms_string.split("/")
+        voms_dict["vo"] = voms_list[1]
+
+        if "Role" not in voms_string:
+            logging.warning(
+                f"No Role found in VOMS of {record.record_id}: {voms_string}"
+            )
+            logging.warning("Not sending VORole")
+            voms_dict["vorole"] = None
+
+            if len(voms_list) == 2:
+                voms_dict["vogroup"] = "/" + voms_list[1]
+            else:
+                voms_dict["vogroup"] = "/" + voms_list[1] + "/" + voms_list[2]
+        elif len(voms_list) == 3:
+            voms_dict["vogroup"] = "/" + voms_list[1]
+            voms_dict["vorole"] = voms_list[2]
+        else:
+            voms_dict["vogroup"] = "/" + voms_list[1] + "/" + voms_list[2]
+            voms_dict["vorole"] = voms_list[3]
+    except TypeError:
+        logging.warning(f"No VOMS information found in {record.record_id}")
+        logging.warning("Not sending VO, VOGroup, and VORole")
```

Comment:
Another one. Btw, this is somewhat a matter of taste. Therefore if you don't agree thats fine.